testsuite/objects/__init__.py

```diff
@@ -76,8 +76,19 @@ class ABCValue(abc.ABC):
     """
     Abstract Dataclass for specifying a Value in Authorization,
     can be either static or reference to value in AuthJson.
+
+    Optional features:
+        - name: for use as JsonProperty
+        - overwrite: for use in ExtendedProperty
+
+    The kw_only argument is used to avoid the need to define default variables in child classes which would otherwise
+    be needed due to how dataclass inheritance works. As these variables are optional it is natural to make them be
+    used as keyword arguments instead of positional arguments.
     """
 
+    name: Optional[str] = field(default=None, kw_only=True)
+    overwrite: Optional[bool] = field(default=None, kw_only=True)
```

Comment:
Having this in an abstract class just for one implementation seems wrong

Reply:
Moved to logic to extra classes as I had previously. I later realized it is a bad idea.